src/main/java/at/hannibal2/skyhanni/features/inventory/ActiveBeaconEffect.kt

```diff
@@ -0,0 +1,49 @@
+package at.hannibal2.skyhanni.features.inventory
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.InventoryUtils.getAllItems
+import at.hannibal2.skyhanni.utils.InventoryUtils.getUpperItems
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.inventory.ContainerChest
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+@SkyHanniModule
+object ActiveBeaconEffect {
+
+    val config get() = SkyHanniMod.feature.inventory
+
+    private val patternGroup = RepoPattern.group("inventory.activebeaconeffect")
+    private val inventoryPattern by patternGroup.pattern(
+        "inventory",
+        "Profile Stat Upgrades"
+    )
+    private val slotPattern by patternGroup.pattern(
+        "slot",
+        "§aActive stat boost!"
+    )
+
+
+    @SubscribeEvent
+    fun onBackgroundDrawn(event: GuiContainerEvent.BackgroundDrawnEvent) {
+        if (!isEnabled()) return
+        if (!inventoryPattern.matches(InventoryUtils.openInventoryName())) return
+
+        val guiChest = event.gui
+        val chest = guiChest.inventorySlots as ContainerChest
+        for ((slot, _) in chest.getUpperItems()) {
+            if (slot.stack.getLore().any { slotPattern.matches(it) }) {
+                slot highlight LorenzColor.GREEN
+                break
+            }
```

Comment:
This logic may be cached on InventoryFullyOpened. But up to you if you want to change it since the performance loss is minimal here anyway in the grand scheme.

Reply:
That's usually what we do. ( Bingo Card Overlay is just a older piece of code).

Reply:
uh so do you have a feature where i can copy from? ;)

Reply:
I have an abstract version in SkyblockGuideHighlightFeature. But there must be other examples somewhere.

Reply:
should i move to a version where the selected one is saved as a slot id? which only gets updated when InventoryFullyOpenedEvent is called?

Reply:
something like that yeah.